src/java/org/apache/cassandra/index/sai/disk/v1/SSTableIndexWriter.java

```diff
@@ -248,8 +240,14 @@ private boolean shouldFlush(long sstableRowId)
 
     private void flushSegment() throws IOException
     {
-        long start = System.nanoTime();
+        currentBuilder.awaitAsyncAdditions();
 
+        // throw exceptions that occurred during async addInternal()
+        var ae = currentBuilder.getAsyncThrowable();
+        if (ae != null)
+            Throwables.throwAsUncheckedException(ae);
```

Comment:
We need to `throw` the exception here

Reply:
Nevermind, Intellij told me the wrong thing

Reply:
yeah, the signature is weird, it should really be void but i decided to leave it alone for now